src/alpaca_parser.yrl

```diff
@@ -145,12 +150,36 @@ record_type_members -> record_type_member ',' record_type_members : ['$1' | '$3'
 record_type -> open_brace record_type_members close_brace : 
   #t_record{members='$2'}.
 
+module_qualified_type_name -> symbol '.' symbol:
+  {symbol, L, Mod} = '$1',
+  {symbol, _, Name} = '$3',
+  {module_qualified_type_name, L, Mod, Name}.
+
+module_qualified_type -> module_qualified_type_name :
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name}}.
+
+module_qualified_type -> module_qualified_type_name type_expressions:
+  {module_qualified_type_name, L, Mod, Name} = '$1',
+  #alpaca_type{
+     line = L,
+     module = list_to_atom(Mod),
+     name = {type_name, L, Name},
+     vars = [V || {type_var, _, _}=V <- '$2']}.
```

Comment:
Your example works because you instantiate the type with a type variable, and those you *do* record on the line above. I meant to highlight that we need to support instantiating with concrete types as well. But I think you maybe addressed that with #113.